osu.Game/Screens/Play/GameplayClockContainer.cs

```diff
@@ -100,6 +100,7 @@ public virtual void Seek(double time)
         /// </summary>
         public virtual void Reset()
         {
+            ChangeSource(SourceClock);
```

Comment:
better with the comment from the other usage:
```suggestion
            // Ensure that the source clock is set.
            ChangeSource(SourceClock);

```

Reply:
would rather factor out a one-liner method at that point if we're nitpicking to this degree.

Reply:
also make the comment mention why it possibly wouldn't be already set. looking at this PR in isolation makes me raise an eyebrow, so a comment may be good to have, explaining what this requirement comes from

Reply:
Have made a change that attempts to explain better. Please advise if it's any better.